packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -102,13 +108,21 @@ export abstract class SharedObject<TEvent extends ISharedObjectEvents = ISharedO
      * @param id - The id of the shared object
      * @param runtime - The IFluidDataStoreRuntime which contains the shared object
      * @param attributes - Attributes of the shared object
+     * @param closeOnEventListenerException - if true (default), results in critical error (container closure) on
+     * exception thrown from from any event listeners for any events raised by this object.
      */
     constructor(
         public id: string,
         protected runtime: IFluidDataStoreRuntime,
-        public readonly attributes: IChannelAttributes)
+        public readonly attributes: IChannelAttributes,
+        closeOnEventListenerException = true)
```

Comment:
I wouldn't even provide an opt-out, if someone screws up "op" processing then we're in danger I think.